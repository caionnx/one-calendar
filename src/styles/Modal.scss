.Modal {
  background: rgb(255, 255, 255);
  border-radius: 4px;
  border: 1px solid rgb(204, 204, 204);
  bottom: auto;
  left: 50%;
  margin-right: -50%;
  outline: none;
  overflow: auto;
  padding: 20px;
  position: absolute;
  right: auto;
  top: 50%;
  transform: translate(-50%, -50%);
  width: 250px;

  select {
    border: 1px solid #333;
    background-color: white;
    outline: none;
  }

  &__Title {
    margin-top: 0;
  }

  &__Flex {
    display: flex;
    justify-content: space-between;
    align-items: flex-start;
  }

  &__Content {
    border-top: 1px solid black;
    border-bottom: 1px solid black;
    padding: 1em 0;
  }
}