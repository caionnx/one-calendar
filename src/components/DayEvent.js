import React, {Component} from 'react';
import Modal from 'react-modal';

import '../styles/Modal.scss';
import '../styles/Button.scss';

class DayEvent extends Component {
  constructor () {
    super();

    this.state = {
      selectedCategory: false,
      defaultCategories: {
        holiday: 'Holiday',
        birthday: 'Birthday',
        busy: 'Busy',
        anniversary: 'Anniversary'
      }
    }
  }

  onChangeCategory (ev) {
    const value = ev.target.value
    this.setState({
      selectedCategory: { id: value, name: this.state.defaultCategories[value] }
    })
  }

  resetSelected () {
    this.setState({
      selectedCategory: false
    })
  }

  render () {
    const {
      modalIsOpen,
      closeModal,
      fullDate,
      event,
      onRemoveEvent,
      onAddEvent,
    } = this.props;

    return (
        <Modal
          className='Modal'
          appElement={document.querySelector('#root')}
          isOpen={modalIsOpen}
          onRequestClose={closeModal}
          contentLabel={fullDate}>

          <div className='Modal__Flex'>
            <h4 className='Modal__Title'>{fullDate}</h4>
            <button
              aria-label='Close modal'
              className='Button Button--Unstyled Button--Large'
              onClick={closeModal}>X</button>
          </div>
          {
            event
            ? <div className='Modal__Content Modal__Flex'>
                {event.name}
                <button className='Button Button--Delete' onClick={() => {
                  this.resetSelected();
                  onRemoveEvent(event.id);
                  }}>
                  Delete event
                </button>
              </div>
            : <form className='Modal__Content Modal__Flex' onSubmit={ev => onAddEvent(ev, this.state.selectedCategory)}>
                <select onChange={ev => this.onChangeCategory(ev)}>
                  <option key='default'>Select one option</option>
                  {Object.keys(this.state.defaultCategories).map(ct => 
                    <option key={ct} value={ct}>{this.state.defaultCategories[ct]}</option>
                  )}
                </select>
                <button className='Button'>Add event</button>
              </form>
          }
        </Modal>
    )
  }
}

export default DayEvent;