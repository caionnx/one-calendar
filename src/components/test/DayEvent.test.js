import React from 'react';
import Modal from 'react-modal';
import { shallow } from 'enzyme';

import DayEvent from '../DayEvent';

let wrapper;

it('render with modal component', () => {
  wrapper = shallow(<DayEvent />);

  expect(wrapper.find(Modal)).toHaveLength(1);
  expect(wrapper).toMatchSnapshot();
});

it('execute "add event" prop function', () => {
  const onAddEvent = jest.fn();
  wrapper = shallow(<DayEvent onAddEvent={onAddEvent} />);
  wrapper.find('select').simulate('change', { target: { value: 'birthday'} })
  wrapper.find('form').simulate('submit');

  expect(onAddEvent).toHaveBeenCalledWith(undefined, {
    id: expect.any(String),
    name: expect.any(String)
  });
});

it('execute "remove event" prop function', () => {
  const onRemoveEvent = jest.fn();
  const event = { name: 'Test', id: 'test' };
  wrapper = shallow(
    <DayEvent event={event} onRemoveEvent={onRemoveEvent} />
  );
  wrapper.find('button').at(1).simulate('click');
  
  expect(wrapper).toMatchSnapshot();
  expect(onRemoveEvent).toHaveBeenCalledWith(event.id);
});