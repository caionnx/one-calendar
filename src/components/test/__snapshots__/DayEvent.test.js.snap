// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`execute "remove event" prop function 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DayEvent
    event={
      Object {
        "id": "test",
        "name": "Test",
      }
    }
    onRemoveEvent={
      [MockFunction] {
        "calls": Array [
          Array [
            "test",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "appElement": null,
      "ariaHideApp": true,
      "bodyOpenClassName": "ReactModal__Body--open",
      "children": Array [
        <div
          className="Modal__Flex Modal__Header"
        >
          <h4
            className="Modal__Title"
          />
          <button
            aria-label="Close modal"
            className="Button Button--Unstyled Button--Large"
          >
            X
          </button>
        </div>,
        <div
          className="Modal__Content Modal__Flex"
        >
          Test
          <button
            className="Button Button--Delete"
            onClick={[Function]}
          >
            Delete event
          </button>
        </div>,
      ],
      "className": "Modal",
      "closeTimeoutMS": 0,
      "contentLabel": undefined,
      "isOpen": false,
      "onRequestClose": undefined,
      "parentSelector": [Function],
      "portalClassName": "ReactModalPortal",
      "role": "dialog",
      "shouldCloseOnEsc": true,
      "shouldCloseOnOverlayClick": true,
      "shouldFocusAfterRender": true,
      "shouldReturnFocusAfterClose": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4
              className="Modal__Title"
            />,
            <button
              aria-label="Close modal"
              className="Button Button--Unstyled Button--Large"
            >
              X
            </button>,
          ],
          "className": "Modal__Flex Modal__Header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "Modal__Title",
            },
            "ref": null,
            "rendered": null,
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-label": "Close modal",
              "children": "X",
              "className": "Button Button--Unstyled Button--Large",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "X",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Test",
            <button
              className="Button Button--Delete"
              onClick={[Function]}
            >
              Delete event
            </button>,
          ],
          "className": "Modal__Content Modal__Flex",
        },
        "ref": null,
        "rendered": Array [
          "Test",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Delete event",
              "className": "Button Button--Delete",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Delete event",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "appElement": null,
        "ariaHideApp": true,
        "bodyOpenClassName": "ReactModal__Body--open",
        "children": Array [
          <div
            className="Modal__Flex Modal__Header"
          >
            <h4
              className="Modal__Title"
            />
            <button
              aria-label="Close modal"
              className="Button Button--Unstyled Button--Large"
            >
              X
            </button>
          </div>,
          <div
            className="Modal__Content Modal__Flex"
          >
            Test
            <button
              className="Button Button--Delete"
              onClick={[Function]}
            >
              Delete event
            </button>
          </div>,
        ],
        "className": "Modal",
        "closeTimeoutMS": 0,
        "contentLabel": undefined,
        "isOpen": false,
        "onRequestClose": undefined,
        "parentSelector": [Function],
        "portalClassName": "ReactModalPortal",
        "role": "dialog",
        "shouldCloseOnEsc": true,
        "shouldCloseOnOverlayClick": true,
        "shouldFocusAfterRender": true,
        "shouldReturnFocusAfterClose": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4
                className="Modal__Title"
              />,
              <button
                aria-label="Close modal"
                className="Button Button--Unstyled Button--Large"
              >
                X
              </button>,
            ],
            "className": "Modal__Flex Modal__Header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "Modal__Title",
              },
              "ref": null,
              "rendered": null,
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "Close modal",
                "children": "X",
                "className": "Button Button--Unstyled Button--Large",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "X",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Test",
              <button
                className="Button Button--Delete"
                onClick={[Function]}
              >
                Delete event
              </button>,
            ],
            "className": "Modal__Content Modal__Flex",
          },
          "ref": null,
          "rendered": Array [
            "Test",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Delete event",
                "className": "Button Button--Delete",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Delete event",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`render with modal component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DayEvent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "appElement": null,
      "ariaHideApp": true,
      "bodyOpenClassName": "ReactModal__Body--open",
      "children": Array [
        <div
          className="Modal__Flex Modal__Header"
        >
          <h4
            className="Modal__Title"
          />
          <button
            aria-label="Close modal"
            className="Button Button--Unstyled Button--Large"
          >
            X
          </button>
        </div>,
        <form
          className="Modal__Content Modal__Flex"
          onSubmit={[Function]}
        >
          <select
            onChange={[Function]}
          >
            <option>
              Select one option
            </option>
            <option
              value="holiday"
            >
              Holiday
            </option>
            <option
              value="birthday"
            >
              Birthday
            </option>
            <option
              value="busy"
            >
              Busy
            </option>
            <option
              value="anniversary"
            >
              Anniversary
            </option>
          </select>
          <button
            className="Button"
          >
            Add event
          </button>
        </form>,
      ],
      "className": "Modal",
      "closeTimeoutMS": 0,
      "contentLabel": undefined,
      "isOpen": false,
      "onRequestClose": undefined,
      "parentSelector": [Function],
      "portalClassName": "ReactModalPortal",
      "role": "dialog",
      "shouldCloseOnEsc": true,
      "shouldCloseOnOverlayClick": true,
      "shouldFocusAfterRender": true,
      "shouldReturnFocusAfterClose": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4
              className="Modal__Title"
            />,
            <button
              aria-label="Close modal"
              className="Button Button--Unstyled Button--Large"
            >
              X
            </button>,
          ],
          "className": "Modal__Flex Modal__Header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "Modal__Title",
            },
            "ref": null,
            "rendered": null,
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-label": "Close modal",
              "children": "X",
              "className": "Button Button--Unstyled Button--Large",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "X",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <select
              onChange={[Function]}
            >
              <option>
                Select one option
              </option>
              <option
                value="holiday"
              >
                Holiday
              </option>
              <option
                value="birthday"
              >
                Birthday
              </option>
              <option
                value="busy"
              >
                Busy
              </option>
              <option
                value="anniversary"
              >
                Anniversary
              </option>
            </select>,
            <button
              className="Button"
            >
              Add event
            </button>,
          ],
          "className": "Modal__Content Modal__Flex",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option>
                  Select one option
                </option>,
                Array [
                  <option
                    value="holiday"
                  >
                    Holiday
                  </option>,
                  <option
                    value="birthday"
                  >
                    Birthday
                  </option>,
                  <option
                    value="busy"
                  >
                    Busy
                  </option>,
                  <option
                    value="anniversary"
                  >
                    Anniversary
                  </option>,
                ],
              ],
              "onChange": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "default",
                "nodeType": "host",
                "props": Object {
                  "children": "Select one option",
                },
                "ref": null,
                "rendered": "Select one option",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "holiday",
                "nodeType": "host",
                "props": Object {
                  "children": "Holiday",
                  "value": "holiday",
                },
                "ref": null,
                "rendered": "Holiday",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "birthday",
                "nodeType": "host",
                "props": Object {
                  "children": "Birthday",
                  "value": "birthday",
                },
                "ref": null,
                "rendered": "Birthday",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "busy",
                "nodeType": "host",
                "props": Object {
                  "children": "Busy",
                  "value": "busy",
                },
                "ref": null,
                "rendered": "Busy",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "anniversary",
                "nodeType": "host",
                "props": Object {
                  "children": "Anniversary",
                  "value": "anniversary",
                },
                "ref": null,
                "rendered": "Anniversary",
                "type": "option",
              },
            ],
            "type": "select",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add event",
              "className": "Button",
            },
            "ref": null,
            "rendered": "Add event",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "appElement": null,
        "ariaHideApp": true,
        "bodyOpenClassName": "ReactModal__Body--open",
        "children": Array [
          <div
            className="Modal__Flex Modal__Header"
          >
            <h4
              className="Modal__Title"
            />
            <button
              aria-label="Close modal"
              className="Button Button--Unstyled Button--Large"
            >
              X
            </button>
          </div>,
          <form
            className="Modal__Content Modal__Flex"
            onSubmit={[Function]}
          >
            <select
              onChange={[Function]}
            >
              <option>
                Select one option
              </option>
              <option
                value="holiday"
              >
                Holiday
              </option>
              <option
                value="birthday"
              >
                Birthday
              </option>
              <option
                value="busy"
              >
                Busy
              </option>
              <option
                value="anniversary"
              >
                Anniversary
              </option>
            </select>
            <button
              className="Button"
            >
              Add event
            </button>
          </form>,
        ],
        "className": "Modal",
        "closeTimeoutMS": 0,
        "contentLabel": undefined,
        "isOpen": false,
        "onRequestClose": undefined,
        "parentSelector": [Function],
        "portalClassName": "ReactModalPortal",
        "role": "dialog",
        "shouldCloseOnEsc": true,
        "shouldCloseOnOverlayClick": true,
        "shouldFocusAfterRender": true,
        "shouldReturnFocusAfterClose": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4
                className="Modal__Title"
              />,
              <button
                aria-label="Close modal"
                className="Button Button--Unstyled Button--Large"
              >
                X
              </button>,
            ],
            "className": "Modal__Flex Modal__Header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "Modal__Title",
              },
              "ref": null,
              "rendered": null,
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "Close modal",
                "children": "X",
                "className": "Button Button--Unstyled Button--Large",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "X",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <select
                onChange={[Function]}
              >
                <option>
                  Select one option
                </option>
                <option
                  value="holiday"
                >
                  Holiday
                </option>
                <option
                  value="birthday"
                >
                  Birthday
                </option>
                <option
                  value="busy"
                >
                  Busy
                </option>
                <option
                  value="anniversary"
                >
                  Anniversary
                </option>
              </select>,
              <button
                className="Button"
              >
                Add event
              </button>,
            ],
            "className": "Modal__Content Modal__Flex",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option>
                    Select one option
                  </option>,
                  Array [
                    <option
                      value="holiday"
                    >
                      Holiday
                    </option>,
                    <option
                      value="birthday"
                    >
                      Birthday
                    </option>,
                    <option
                      value="busy"
                    >
                      Busy
                    </option>,
                    <option
                      value="anniversary"
                    >
                      Anniversary
                    </option>,
                  ],
                ],
                "onChange": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "default",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Select one option",
                  },
                  "ref": null,
                  "rendered": "Select one option",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "holiday",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Holiday",
                    "value": "holiday",
                  },
                  "ref": null,
                  "rendered": "Holiday",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "birthday",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Birthday",
                    "value": "birthday",
                  },
                  "ref": null,
                  "rendered": "Birthday",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "busy",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Busy",
                    "value": "busy",
                  },
                  "ref": null,
                  "rendered": "Busy",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "anniversary",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Anniversary",
                    "value": "anniversary",
                  },
                  "ref": null,
                  "rendered": "Anniversary",
                  "type": "option",
                },
              ],
              "type": "select",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add event",
                "className": "Button",
              },
              "ref": null,
              "rendered": "Add event",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
